apps/system/js/lock_screen_window_manager.js

```diff
@@ -240,7 +240,15 @@
       // Should be replaced by a proper IAC API in bug 992277.
       var lockListener = (event) => {
         if (true === event.settingValue) {
-          this.openApp();
+          // Ignore when lockscreen is disabled.
+          // Prevents LSWM ending up in a broken, permanently-locked
+          // state when lock-immediately is sent to disabled lockscreen.
+          if (this.states.enabled) {
```

Comment:
I think for FMD it need to lock it forcibly no matter whether the enable is true? Or a thief can easily bypass that when the settings is still showing on the screen. And the issue of unable to unlock the phone again after exiting FMD mode as you describe here should be fixed in another bug if we didn't fire one.